### PR TITLE
Adding info about quorum-aware data-structures

### PR DESCRIPTION
**Pull request by tombujok**

### CODE REVIEW

File: cluster-quorum-xml/src/main/java/ClusterQuorum.java [hunk 1]
```diff
@@ -4,6 +4,34 @@
 import com.hazelcast.core.IQueue;
 import com.hazelcast.quorum.QuorumException;
 
+/**
+ * The term "quorum" simply refers to the count of members in the cluster required for an operation to succeed.
+ * It does NOT refer to an implementation of Paxos or Raft protocols as used in many NoSQL and distribuetd systems.
+ * The mechanism it provides in Hazelcast protects the user in case the number of nodes in a cluster drops below the specified one.
+ *
+ * Hazelcast Quorum is supported in the following data-structures:
+ * - Map
```
Comment (Donnerbart) [inline cluster-quorum-xml/src/main/java/ClusterQuorum.java]: Should be `IMap`

Comment (Donnerbart) [reply cluster-quorum-xml/src/main/java/ClusterQuorum.java]: Also `ICache`, `IQueue` and `ILock`.

Comment (tombujok) [reply cluster-quorum-xml/src/main/java/ClusterQuorum.java]: Thx, I also fixed some others, like RingBuffer (not IRingbuffer, etc.)

Comment (Donnerbart) [reply cluster-quorum-xml/src/main/java/ClusterQuorum.java]: Good catch, but actually it's `Ringbuffer` not `RingBuffer` :D

File: cluster-quorum-xml/src/main/resources/hazelcast.xml [hunk 2]
```diff
@@ -14,4 +22,30 @@
...
+         - ISemaphore
+         - ICountDownLatch
+         - IAtomicLong
+         - IAtomicReference
+         - ReplicatedMap
+         - MultiMap
+         - IExecutorService
+         - DurableExecutorService
+         - IScheduledExecutorService
+         - RingBuffer
```
[the remark this answers is not in the document]
Comment (tombujok) [reply cluster-quorum-xml/src/main/resources/hazelcast.xml]: thx :)

**Pull request merged**